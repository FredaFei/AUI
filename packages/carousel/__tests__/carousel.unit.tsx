import * as renderer from 'react-test-renderer'
import React from 'react'
import {mount} from 'enzyme'

import Carousel from '../index'
import CarouselItem from '../carouselItem'

describe('测试 Carousel 以下行为', () => {
  it('Carousel 存在', () => {
    const json = renderer.create(<Carousel style={{ height: 300 }}>
      <CarouselItem style={{ background: '#66e2d5', color: '#fff', fontSize: '20px' }}>page 1</CarouselItem>
      <CarouselItem style={{ background: 'yellow', color: '#fff', fontSize: '20px' }}>page 2</CarouselItem>
    </Carousel>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('Carousel 默认属性 animation/autoPlay/duration/indicator', () => {
    const wrapper = mount(<Carousel style={{ height: 300 }}>
      <CarouselItem style={{ background: '#66e2d5', color: '#fff', fontSize: '20px' }}>page 1</CarouselItem>
      <CarouselItem style={{ background: 'yellow', color: '#fff', fontSize: '20px' }}>page 2</CarouselItem>
    </Carousel>)
    expect(wrapper.props().animation).toEqual('slide')
    expect(wrapper.props().autoPlay).toEqual(false)
    expect(wrapper.props().duration).toEqual(2000)
    expect(JSON.stringify(wrapper.props().indicator)).toEqual(JSON.stringify({ position: 'bottom', style: 'circle' }))
  })
  it('Carousel 设置 autoPlay', () => {
    const wrapper = mount(<Carousel style={{ height: 300 }} autoPlay>
      <CarouselItem style={{ background: '#66e2d5', color: '#fff', fontSize: '20px' }}>page 1</CarouselItem>
      <CarouselItem style={{ background: 'yellow', color: '#fff', fontSize: '20px' }}>page 2</CarouselItem>
    </Carousel>)
    expect(wrapper.props().autoPlay).toEqual(true)
  })
  it('Carousel 设置 indicator', () => {
    const wrapper = mount(<Carousel style={{ height: 300 }} indicator={{ position: 'left', style: 'number' }}>
      <CarouselItem style={{ background: '#66e2d5', color: '#fff', fontSize: '20px' }}>page 1</CarouselItem>
      <CarouselItem style={{ background: 'yellow', color: '#fff', fontSize: '20px' }}>page 2</CarouselItem>
    </Carousel>)
    console.log(wrapper.props().indicator);
    expect(JSON.stringify(wrapper.props().indicator)).toEqual(JSON.stringify({ position: 'left', style: 'number' }))
  })
  it('Carousel 设置 duration', () => {
    const wrapper = mount(<Carousel style={{ height: 300 }} duration={3000}>
      <CarouselItem style={{ background: '#66e2d5', color: '#fff', fontSize: '20px' }}>page 1</CarouselItem>
      <CarouselItem style={{ background: 'yellow', color: '#fff', fontSize: '20px' }}>page 2</CarouselItem>
    </Carousel>)
    expect(wrapper.props().duration).toEqual(3000)
  })
  it('Carousel 设置 animation', () => {
    const wrapper = mount(<Carousel style={{ height: 300 }} animation="fade">
      <CarouselItem style={{ background: '#66e2d5', color: '#fff', fontSize: '20px' }}>page 1</CarouselItem>
      <CarouselItem style={{ background: 'yellow', color: '#fff', fontSize: '20px' }}>page 2</CarouselItem>
    </Carousel>)
    expect(wrapper.props().animation).toEqual('fade')
  })
})