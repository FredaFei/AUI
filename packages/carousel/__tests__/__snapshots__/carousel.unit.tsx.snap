// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Carousel 以下行为 Carousel 存在 1`] = `
<div
  className="am-carousel-wrapper "
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="am-carousel-view am-carousel-slide am-carousel"
    style={
      Object {
        "height": 300,
      }
    }
  >
    <div
      className="am-carouselitem am-carouselitem-current "
      style={
        Object {
          "background": "#66e2d5",
          "color": "#fff",
          "fontSize": "20px",
        }
      }
    >
      page 1
    </div>
    <div
      className="am-carouselitem "
      style={
        Object {
          "background": "yellow",
          "color": "#fff",
          "fontSize": "20px",
        }
      }
    >
      page 2
    </div>
  </div>
  <div
    className="am-carousel-dots position-bottom style-circle"
  >
    <span
      className="dot position-bottom style-circle active"
      onClick={[Function]}
    >
      
    </span>
    <span
      className="dot position-bottom style-circle "
      onClick={[Function]}
    >
      
    </span>
  </div>
</div>
`;
